### PR TITLE
Extend number parsing to work for all radix

### PR DESCRIPTION
**Pull request by lask**

### CODE REVIEW

File: lib/core/numbers.toit [hunk 1]
```diff
@@ -616,7 +616,28 @@ abstract class int extends num:
     else if radix == 16:
       return parse_16_ data from to --on_error=on_error
     else:
-      return on_error.call PARSE_ERR_
+      return parse_generic_radix radix data from to --on_error=on_error
+
+  static parse_generic_radix radix/int data from/int to/int [--on_error] -> int?:
+    if radix <= 1 or radix > 36: throw PARSE_ERR_
```
Comment (floitsch) [inline lib/core/numbers.toit:622]:
```suggestion
    if not 2 <= radix <= 36: throw "INVALID RADIX"
```

File: lib/core/numbers.toit [hunk 2]
```diff
@@ -561,7 +561,7 @@ abstract class int extends num:
 
   The data must be a $string or $ByteArray.
 
-  The given $radix must be either 10 (default) or 16.
+  The given $radix must be in the range 2 to 36.
```
Comment (floitsch) [inline lib/core/numbers.toit:564]:
```suggestion
  The given $radix must be in the range 2 to 36 (inclusive).
```

File: lib/core/numbers.toit [hunk 3]
```diff
@@ -616,7 +616,28 @@ abstract class int extends num:
...
-      return on_error.call PARSE_ERR_
+      return parse_generic_radix radix data from to --on_error=on_error
+
+  static parse_generic_radix radix/int data from/int to/int [--on_error] -> int?:
+    if radix <= 1 or radix > 36: throw PARSE_ERR_
+
+    max_num := (min radix 9) + '0' - 1
+    max_char := radix + 'a' - 1
+    max_char_C := radix + 'A' - 1
+    print "radix: $radix max_num: $max_num max_char: $max_char max_char_C: $max_char_C"
```
Comment (floitsch) [inline lib/core/numbers.toit:627]:
```suggestion
```

**Review by floitsch — APPROVED** [review]
LGTM with some tests.

**Comment by kasperl** [comment]
The new functionality landed in 4eddb1bb79ba32e9ec2d80e287ba4f2019b75c36.

**Pull request closed without merging**